src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +24,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle current = (Rectangle) rectangle;
+            return Objects.equals(this.width, current.width)
+                  && Objects.equals(this.length, current.length)
+                  && Objects.equals(this.color, current.color);
+        }
```

Comment:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/equals-and-hashcode/equals-and-hashcode